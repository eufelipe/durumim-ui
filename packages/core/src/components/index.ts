export * from './button/button';
export * from './typography/typography';