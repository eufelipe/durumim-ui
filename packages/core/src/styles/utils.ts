import { StyleRule } from '@vanilla-extract/css';

export type ResponsiveValue<Value> = Value | Partial<Record<'sm' | 'md' | 'lg' | 'xl' | '2xl', Value>>;

export const mapResponsiveValue = <Value>(
  value: ResponsiveValue<Value>,
  mapFn: (value: Value) => StyleRule
): StyleRule => {
  if (typeof value !== 'object' || value === null) {
    return mapFn(value as Value);
  }

  const breakpointEntries = Object.entries(value as Record<string, Value>);

  if (breakpointEntries.length === 0) {
    return {};
  }

  const mediaQueries: Record<string, StyleRule> = {};

  for (const [breakpoint, breakpointValue] of breakpointEntries) {
    if (breakpoint === 'base') {
      Object.assign(mediaQueries, mapFn(breakpointValue));
    } else {
      mediaQueries[`@media screen and (min-width: ${getBreakpointValue(breakpoint)})`] = mapFn(breakpointValue);
    }
  }

  return mediaQueries;
};

const getBreakpointValue = (breakpoint: string): string => {
  switch (breakpoint) {
    case 'sm':
      return '640px';
    case 'md':
      return '768px';
    case 'lg':
      return '1024px';
    case 'xl':
      return '1280px';
    case '2xl':
      return '1536px';
    default:
      return '0';
  }
}; 