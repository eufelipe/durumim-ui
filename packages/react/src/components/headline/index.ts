export * from "./Headline1";
export * from "./Headline2";
export * from "./Headline3";
export * from "./Headline4";
export * from "./Headline5";
export * from "./Title";
