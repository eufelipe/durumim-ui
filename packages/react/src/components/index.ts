export * from "./headline";
export * from "./text";
