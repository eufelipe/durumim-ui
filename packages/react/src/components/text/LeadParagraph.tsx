import { ElementType } from "react";

import { CSSProps, css, styled } from "@durumim-ui/core";

const styles = css({
  fontFamily: "$default",
  fontSize: "$lg",
  lineHeight: "$base",
  margin: 0,
  color: "$gray100",
});

const LeadParagraphComponent = styled("p", styles);

export interface LeadParagraphProps extends CSSProps {
  as?: ElementType;
}
export const LeadParagraph: React.FC<LeadParagraphProps> = ({
  as,
  ...props
}) => {
  return <LeadParagraphComponent as={as} {...props} />;
};

LeadParagraph.displayName = "LeadParagraph";
