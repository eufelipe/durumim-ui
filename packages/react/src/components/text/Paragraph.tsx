import { ElementType } from "react";

import { CSSProps, css, styled } from "@durumim-ui/core";

const styles = css({
  fontFamily: "$default",
  fontSize: "$md",
  lineHeight: "$base",
  margin: 0,
  color: "$gray100",
});

const ParagraphComponent = styled("p", styles);

export interface ParagraphProps extends CSSProps {
  as?: ElementType;
}
export const Paragraph: React.FC<ParagraphProps> = ({ as, ...props }) => {
  return <ParagraphComponent as={as} {...props} />;
};

Paragraph.displayName = "Paragraph";
