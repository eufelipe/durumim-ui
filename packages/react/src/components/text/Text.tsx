import { ElementType } from "react";

import { CSSProps, css, styled } from "@durumim-ui/core";

const styles = css({
  fontFamily: "$default",
  lineHeight: "$base",
  margin: 0,
  color: "$gray100",

  variants: {
    size: {
      xxs: { fontSize: "$xxs" },
      xs: { fontSize: "$xs" },
      sm: { fontSize: "$sm" },
      md: { fontSize: "$md" },
      lg: { fontSize: "$lg" },
      xl: { fontSize: "$xl" },
      "2xl": { fontSize: "$2xl" },
      "4xl": { fontSize: "$4xl" },
      "5xl": { fontSize: "$5xl" },
      "6xl": { fontSize: "$6xl" },
      "7xl": { fontSize: "$7xl" },
      "8xl": { fontSize: "$8xl" },
      "9xl": { fontSize: "$9xl" },
    },
  },

  defaultVariants: {
    size: "md",
  },
});

const TextComponent = styled("p", styles);

export interface TextProps extends CSSProps {
  as?: ElementType;
}
export const Text: React.FC<TextProps> = ({ as, ...props }) => {
  return <TextComponent as={as} {...props} />;
};

Text.displayName = "Text";
