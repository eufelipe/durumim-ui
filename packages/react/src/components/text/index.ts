export * from "./LeadParagraph";
export * from "./Text";
