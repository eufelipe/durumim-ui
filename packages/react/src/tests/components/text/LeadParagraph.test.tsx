import { describe, expect, it } from "vitest";

import { LeadParagraph } from "@/components/text";
import { render, screen } from "@/tests/utils";

describe("<LeadParagraph />", () => {
  beforeEach(() => {
    render(
      <LeadParagraph data-testid="lead-paragraph">
        Você, precisa aprender inglês, precisa aprender o que eu sei
      </LeadParagraph>
    );
  });

  it("should render the lead paragraph correctly as expected", () => {
    expect(
      screen.getByText(
        /Você, precisa aprender inglês, precisa aprender o que eu sei/i
      )
    ).toBeInTheDocument();
  });

  it("shouldn't render the lead paragraph correctly", () => {
    expect(
      screen.queryByText(/E o que eu não sei mais/i)
    ).not.toBeInTheDocument();
  });

  it("should render correctly", () => {
    expect(screen.getByTestId("lead-paragraph")).toMatchSnapshot();
  });
});
