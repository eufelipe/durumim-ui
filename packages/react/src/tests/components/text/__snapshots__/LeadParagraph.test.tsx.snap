// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<LeadParagraph /> > should render correctly 1`] = `
<h6
  class="c-kfKdCb"
  data-testid="lead-paragraph"
>
  Você, precisa aprender inglês, precisa aprender o que eu sei
</h6>
`;
